### PR TITLE
add rule prefer-shorthand-css-notations

### PR DESCRIPTION
**Pull request by elisecodedestrucs**
add a rule that aims to reduce stylesheet size by using shorthand css notations

### CODE REVIEW

**Review by utarwyn — COMMENTED** [review]
Thank you for the rule @elisecodedestrucs!
I have a question: in the proposed solution, it's not possible to have "margin-top" or a "padding-bottom" alone, for example? (a warning will be raised) I don't think it's bad practice to specify just one of the margins/borders using the dedicated rule. What do you think of this?

File: eslint-plugin/tests/lib/rules/prefer-shorthand-css-notations.js
Comment (utarwyn) [inline eslint-plugin/tests/lib/rules/prefer-shorthand-css-notations.js:1]: This file seems to be missing a few tests in order to cover all CSS properties?

**Comment by elisecodedestrucs** [comment]
> Thank you for the rule @elisecodedestrucs! I have a question: in the proposed solution, it's not possible to have "margin-top" or a "padding-bottom" alone, for example? (a warning will be raised) I don't think it's bad practice to specify just one of the margins/borders using the dedicated rule. What do you think of this?

We agree the rule might have been a little too restrictive, to fix this we've sepparated it between five rules that each cover a few css properties we considered fairly related, the goal is to be able to unable one or some rules if they turn out to be inadapted
 we've also tried to tailor the rule's comportment to the possibilities of each shorthand notation

File: eslint-plugin/lib/rules/prefer-shorthand-css-notations-margin-padding.js
Comment (utarwyn) [inline eslint-plugin/lib/rules/prefer-shorthand-css-notations-margin-padding.js:1]: I think we could factorize checks done for this rule, but for a V1 it's fine with me 👍

File: eslint-plugin/tests/lib/rules/prefer-shorthand-css-notations-margin-padding.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+          data: {
+            attribute: "margin",
+            value: `margin: 10 8 7`,
+          },
+          type: "JSXAttribute",
+        },
+      ],
+    },
+    {
+      code: "<div style={{ paddingTop: 10, paddingBottom: 8, paddingRight:3, paddingLeft:5 }}> {/* Your content here */} </div>",
```
Comment (utarwyn) [inline eslint-plugin/tests/lib/rules/prefer-shorthand-css-notations-margin-padding.js:100]: It seems to miss some tests for padding properties with the same value for top/bottom and left/right (like you have done with margin properties)

**Review by utarwyn — APPROVED** [review]

**Pull request merged**